tests/generic_config_updater/test_pfcwd_interval.py

```diff
@@ -0,0 +1,135 @@
+            interface = line.split()[0] #Since line starts with Ethernet, we can safely use 0 index
+
+            cmd = "sonic-db-cli CONFIG_DB hget \"PFC_WD|{}\" \"detection_time\" ".format(interface)
+            output = duthost.shell(cmd, module_ignore_errors=True)
+            pytest_assert(not output['rc'], "Unable to read detection time")
+            detection_time = output["stdout"]
+            
+            cmd = "sonic-db-cli CONFIG_DB hget \"PFC_WD|{}\" \"restoration_time\" ".format(interface)
+            output = duthost.shell(cmd, module_ignore_errors=True)
+            pytest_assert(not output['rc'], "Unable to read restoration time")
+            restoration_time = output["stdout"]
+
+            return int(detection_time), int(restoration_time)
+
+    pytest_assert(True, "Failed to read detection_time and/or restoration time")
+            
+
+def get_new_interval(duthost, is_valid):
+    """
+    Returns new interval value for pfcwd poll interval, based on the operation being performed
+
+    Args:
+        duthost: DUT host object
+        is_valid: if is_valid is true, return a valid new interval. Config update should succeed. If is_valid is false, return an invalid new interval. Config update should fail.
+    """
+
+    detection_time, restoration_time = get_detection_restoration_times(duthost)
+    if is_valid:
+        return max(detection_time, restoration_time) + 10
+    else:
+        return min(detection_time, restoration_time) - 10
+
+
+@pytest.mark.parametrize("operation", ["add", "replace", "remove"])
+@pytest.mark.parametrize("field_pre_status", ["existing", "nonexistent"])
+@pytest.mark.parametrize("is_valid_config_update", [True, False])
+def test_pfcwd_interval_config_updates(duthost, ensure_dut_readiness, operation, field_pre_status, is_valid_config_update):
+    new_interval = get_new_interval(duthost, is_valid_config_update)
+
+    operation_to_new_value_map = {"add": "{}".format(new_interval), "replace": "{}".format(new_interval), "remove": ""}
+    detection_time, restoration_time = get_detection_restoration_times(duthost)
+    pre_status = max(detection_time, restoration_time)
+    field_pre_status_to_value_map = {"existing": "{}".format(pre_status), "nonexistent": ""}
+    
+    prepare_pfcwd_interval_config(duthost, field_pre_status_to_value_map[field_pre_status]) 
+
+    tmpfile = generate_tmpfile(duthost)
+    logger.info("tmpfile {} created for json patch of pfcwd poll interval and operation: {}".format(tmpfile, operation))
+    value = operation_to_new_value_map[operation]
+    logger.info("value to be added to json patch: {}".format(value))
+
+    json_patch = [
+        {
+            "op": "{}".format(operation), 
+            "path": "/PFC_WD/GLOBAL/POLL_INTERVAL", 
+            "value": "{}".format(value)
+        }]
+    
+    output = apply_patch(duthost, json_data=json_patch, dest_file=tmpfile)
+    
+    if is_valid_config_update:
+        expect_op_success(duthost, output)
```

Comment:
>expect_op_success [](http://example.com/codeflow?start=8&length=17)

You only validate the return code of `apply_patch`. Do we need more validation? like overall health, no crash, etc.

Reply:
Yes, added check to read from value `FLEX_COUNTER_GROUP_TABLE` which is populated upon successful update of `POLL_INTERVAL`